commit 68e8f9ae59af05c395706a7dbb2451f0b3f4b4fc
Author: QbProg <tholag@gmail.com>
Date:   Sat Jun 11 20:07:46 2011 +0200

    Fixed or marked some bugs thanks to warning messages.
    This includes a buffer overflow in Interface_MSG

Index: oce/src/Interface/Interface_MSG.cxx
===================================================================
--- oce.orig/src/Interface/Interface_MSG.cxx
+++ oce/src/Interface/Interface_MSG.cxx
@@ -59,7 +59,7 @@ Interface_MSG::Interface_MSG
 {
   char mess[300];
   sprintf (mess, Interface_MSG::Translated(thekey), i1);
-  theval = new char (strlen (mess)+1 );
+  theval = new char [strlen (mess)+1 ];
   strcpy (theval,mess);
 }
 
@@ -70,7 +70,7 @@ Interface_MSG::Interface_MSG
 {
   char mess[300];
   sprintf (mess, Interface_MSG::Translated(thekey), i1,i2);
-  theval = new char (strlen (mess)+1 );
+  theval = new char [strlen (mess)+1 ];
   strcpy (theval,mess);
 }
 
@@ -82,7 +82,7 @@ Interface_MSG::Interface_MSG
   char mess[300];
   sprintf (mess, Interface_MSG::Translated(thekey),
 	   (intervals < 0 ? r1 : Interface_MSG::Intervalled(r1,intervals)) );
-  theval = new char (strlen (mess)+1 );
+  theval = new char [strlen (mess)+1 ];
   strcpy (theval,mess);
 }
 
@@ -93,7 +93,7 @@ Interface_MSG::Interface_MSG
 {
   char mess[300];
   sprintf (mess, Interface_MSG::Translated(thekey), str);
-  theval = new char (strlen (mess)+1 );
+  theval = new char [strlen (mess)+1];
   strcpy (theval,mess);
 }
 
@@ -105,7 +105,7 @@ Interface_MSG::Interface_MSG
 {
   char mess[300];
   sprintf (mess, Interface_MSG::Translated(thekey), val, str);
-  theval = new char (strlen (mess)+1 );
+  theval = new char [strlen (mess)+1 ];
   strcpy (theval,mess);
 }
 
